codec-http2/src/main/java/io/netty/handler/codec/http2/DefaultHttp2ConnectionEncoder.java

```diff
@@ -246,7 +246,8 @@ public ChannelFuture writePushPromise(ChannelHandlerContext ctx, int streamId, i
 
             Http2Stream stream = requireStream(streamId);
             // Reserve the promised stream.
-            connection.local().reservePushStream(promisedStreamId, stream);
+            Http2Stream reservedStream = connection.local().reservePushStream(promisedStreamId, stream);
+            reservedStream.headerSent();
```

Comment:
can you comment as to why this is needed? The state will not be idle, and the frame will not be queued in the flow controller.